src/app/domain/authentication/models/User.ts

```diff
@@ -0,0 +1,137 @@
+import CozyClient, { Q } from 'cozy-client'
+import { ClientCapabilities } from 'cozy-client/types/types'
+import flag from 'cozy-flags'
+
+import { devlog } from '/core/tools/env'
+
+export const canAuthWithOIDC = (client: CozyClient): boolean | undefined => {
+  return isOidcAuth(client)
+}
+
+export const shouldCreatePassword = async (
```

Comment:
nit: As we are in a generic `User` model, I think the `shouldCreatePassword` is too specific to your onboarding prompt. I would prefer a `hasDefinedPassword` that only describe the user's state

Note that this would reverse the return statement:
```js
const passwordDefined = await isPasswordDefined(client)
const vaultInstalled = await IsVaultInstalled(client)

return (
    vaultInstalled || passwordDefined || (!isOidcAuth(client) && !isMagicLinkAuth(client))
)
```

This can be simplified with:
```js
return (
    vaultInstalled || passwordDefined || (!isPasswordlessCozy(client))
)

//...

const isPasswordLessCozy = (client) => {
    return isOidcAuth(client) || isMagicLinkAuth(client)
}
```

Reply:
- [ ] Updated